devito/interfaces.py

```diff
@@ -127,7 +132,10 @@ def set_initializer(self, lambda_initializer):
         self.initializer = lambda_initializer
 
     def _allocate_memory(self):
-        self._data = aligned(np.zeros(self.shape, self.dtype, order='C'), alignment=64)
+        if self.memmap:
+            self._data = aligned(np.memmap(filename=(self.f), dtype=self.dtype, mode='w+', shape=tuple(self.shape), order='C'), alignment=64)
```

Comment:
Since I wrote the `aligned` function thinking of numpy `ndarray`s, I doubt it will work with numpy `memmap`s. Reading their implementation only strengthens my doubt. `memmap` doesn't seem to own its block of memory which is deferred to `mmap`. I doubt that `aligned` will be able to reallocate it. Worse, it might bypass the complete `memmap` functionality and return a pure `ndarray` object. I suggest you do not let this call to `aligned` stay unless you are sure it's doing what you expect it to do. The call in the `else`, which was the line of code in master stays of course.